server/src/main/java/org/apache/iotdb/db/engine/storagegroup/StorageGroupProcessor.java

```diff
@@ -1443,13 +1443,15 @@ private void logDeletion(long startTime, long endTime, PartialPath deviceId, Str
 
   private void deleteDataInFiles(Collection<TsFileResource> tsFileResourceList, Deletion deletion,
       List<ModificationFile> updatedModFiles)
-      throws IOException {
-    String deviceId = deletion.getDevice();
+          throws IOException, MetadataException {
     for (TsFileResource tsFileResource : tsFileResourceList) {
-      if (!tsFileResource.containsDevice(deviceId) ||
-          deletion.getEndTime() < tsFileResource.getStartTime(deviceId) ||
-          deletion.getStartTime() > tsFileResource.getOrDefaultEndTime(deviceId, Long.MAX_VALUE)) {
-        continue;
+      for (PartialPath p : IoTDB.metaManager.getAllTimeseriesPath(deletion.getPath())) {
+        String deviceId = p.getDevice();
+        if (!tsFileResource.containsDevice(deviceId) ||
+                deletion.getEndTime() < tsFileResource.getStartTime(deviceId) ||
+                deletion.getStartTime() > tsFileResource.getOrDefaultEndTime(deviceId, Long.MAX_VALUE)) {
+          continue;
+        }
```

Comment:
I think you also write every mod record into tsfile resource's mod file. In jira, you said that 
```
I will maintain a global modification file for one storage group. When query, the executor will check the file to filter the series.
```
Could you explain that?   Thanks~

Reply:
Sure! In previous implementation, only timeseries fullpath is written into mod file. 
E.g. the deletion path is "_root.sg.d1.*_" and d1 has N measurements, we have to access mod file N times to record all the fullpaths.
For now, we record "_root.sg.d1.*_" into mod file only once.

Reply:
The for (PartialPath p ) statement is of no use. Check the code carefully.

insert into root.turbine1.a.d1(timestamp,s1) values(1,1);
insert into root.turbine1.a.d2(timestamp,s1) values(1,1);
flush
insert into root.turbine1.b.d1(timestamp,s1) values(2,1);
insert into root.turbine1.b.d2(timestamp,s1) values(2,1);
flush
delete from root.turbine1 where time < 5

You will record all files a mod.

Reply:
That's what we expected. 
For one delete time series, record one mod information for all mod files, but only one record per file.